tasks/pipeline.py

```diff
@@ -463,6 +465,94 @@ def trigger_child_pipeline(_, git_ref, project_name, variables="", follow=True):
+        if len(match_teams) != 0:
+            return True
+
+    return False
+
+
+@task
+def changelog(ctx, new_commit_sha):
+    old_commit_sha = ctx.run(
+        "aws ssm get-parameter --region us-east-1 --name "
+        "ci.datadog-agent.gitlab_changelog_commit_sha --with-decryption --query "
+        "\"Parameter.Value\" --out text",
+        hide=True,
+    ).stdout.strip()
+    if not new_commit_sha:
+        print("New commit sha not found, exiting")
+        return
+    if not old_commit_sha:
+        print("Old commit sha not found, exiting")
+        return
+    print(f"Generating changelog for commit range {old_commit_sha} to {new_commit_sha}")
+    commits = ctx.run(f"git log {old_commit_sha}..{new_commit_sha} --pretty=format:%h", hide=True).stdout.split("\n")
+    owners = read_owners(".github/CODEOWNERS")
+    messages = []
+
+    for commit in commits:
+        # see https://git-scm.com/docs/pretty-formats for format string
+        commit_str = ctx.run(f"git show --name-only --pretty=format:%s%n%aN%n%aE {commit}", hide=True).stdout
+        title, author, author_email, files, url = parse(commit_str)
+        if is_system_probe(owners, files):
+            author_handle = ctx.run(f"email2slackid {author_email.strip()}", hide=True).stdout
+            if author_handle:
+                author_handle = f"<@{author_handle}>"
+            else:
+                author_handle = author_email
+            time.sleep(1)  # necessary to prevent slack/sdm API rate limits
+            message_link = f"• <{url}|{title}>"
+            if "dependabot" not in author_email and "github-actions" not in author_email:
+                messages.append(f"{message_link} {author_handle}")
+            else:
+                messages.append(f"{message_link}")
+
+    commit_range_link = f"https://github.com/DataDog/datadog-agent/compare/{old_commit_sha}..{new_commit_sha}"
+    slack_message = f"The nightly deployment is rolling out to Staging :siren: \n"
+    if messages:
+        slack_message += (
+            f"Changelog for commit <{commit_range_link}|range>: `{old_commit_sha}` to `{new_commit_sha}`:\n"
+            + "\n".join(messages)
+            + "\n:wave: Authors, please check relevant "
+            "<https://ddstaging.datadoghq.com/dashboard/kfn-zy2-t98|dashboards> for issues"
+        )
+    else:
+        slack_message += (
+            f"Changelog for commit <{commit_range_link}|range>: `{old_commit_sha}` to `{new_commit_sha}`:\n"
+            + "No new System Probe commits in this release :cricket:"
+        )
+    print(f"Pushing new tag: {new_commit_sha} to SSM")
+    ctx.run(
+        f"aws ssm put-parameter --name ci.datadog-agent.gitlab_changelog_commit_sha --value {new_commit_sha} "
+        "--type \"SecureString\" --region us-east-1 --overwrite",
+        hide=True,
+    )
+    print(f"Posting message to slack \n {slack_message}")
+    send_slack_message("system-probe-ops", slack_message)
```

Comment:
What happens if the sending of slack message fails?

Reply:
I'll change this at least to try posting to slack before updating state so it can be rerun manually